File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
      unordered_set<int>s(nums1.begin(),nums1.end());
      unordered_set<int>ans;
      for(int i:nums2)
      {
        if(s.count(i))
        ans.insert(i);
      } 
      return vector<int>(ans.begin(),ans.end()); 
    }
};

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer>st=new HashSet<>();
        List<Integer>result=new ArrayList<>();
        for(int num:nums1)
        {
            st.add(num);
        }
        for(int num:nums2)
        {
            if(st.contains(num))
            {
                result.add(num);
                st.remove(num);
            }
        }
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
}